<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Integration of Seurat objects â€¢ seuFLViz</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Integration of Seurat objects">
<meta property="og:description" content="seuFLViz">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">seuFLViz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.6.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/integration.html">Integration of Seurat objects</a>
    </li>
    <li>
      <a href="../articles/preprocessing.html">Preprocessing</a>
    </li>
    <li>
      <a href="../articles/shiny_app.html">Shiny App</a>
    </li>
    <li>
      <a href="../articles/visualization.html">Visualization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cobriniklab/seuFLViz/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Integration of Seurat objects</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/cobriniklab/seuFLViz/blob/HEAD/vignettes/articles/integration.Rmd" class="external-link"><code>vignettes/articles/integration.Rmd</code></a></small>
      <div class="hidden name"><code>integration.Rmd</code></div>

    </div>

    
    
<p>When joint analysis of two or more single cell sequencing data is to
be performed, we need to first integrate the individual datasets. In
seuFLViz, a single function <code>integration_workflow</code> is
capable of integrating multiple Seurat objects provided as a list.</p>
<p>First step in this process is to load all the packages required</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cobriniklab/seuFLViz" class="external-link">seuFLViz</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com" class="external-link">ggraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="tldr">TLDR<a class="anchor" aria-label="anchor" href="#tldr"></a>
</h2>
<p>Here, we are using the <code>panc8</code> dataset, which is produced
in two batches using different technologies. The
<code><a href="https://satijalab.org/seurat/reference/SplitObject.html" class="external-link">SplitObject()</a></code> function splits the seurat object into a list
containing each batch as element. This list of seurat objects can then
be integrated using the <code><a href="../reference/integration_workflow.html">integration_workflow()</a></code> function,
which identifies shared cell states that are present across different
single cell datasets.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">batches</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">human_gene_transcript_seu</span><span class="op">[</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span><span class="op">$</span><span class="va">Prep.Method</span><span class="op">)</span><span class="op">]</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SplitObject.html" class="external-link">SplitObject</a></span><span class="op">(</span>split.by <span class="op">=</span> <span class="st">"Prep.Method"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/identity.html" class="external-link">identity</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">integrated_seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/integration_workflow.html">integration_workflow</a></span><span class="op">(</span><span class="va">batches</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="batch-effects-are-problematic">Batch effects are problematic<a class="anchor" aria-label="anchor" href="#batch-effects-are-problematic"></a>
</h2>
<p>When analyzing data across multiple batches, batch effects can become
problematic and compromise integration and interpretation of the data.
For instance, the dataset <code>human_gene_transcript_seu</code>
contains data obtained following two different preparation method. When
this data is analyzed without integration we notice that in the Zhong
method cluster 2 is barely represented. from the plot of marker features
we know that cluster 2 has high expression of features associated with
cones. this absence of cluster 2 in in Zhong method can become
problematic during downstream analysis</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markerp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_markers.html">plot_markers</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span>, metavar <span class="op">=</span> <span class="st">"gene_snn_res.0.2"</span>, num_markers <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">dp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span>, group.by <span class="op">=</span> <span class="st">"gene_snn_res.0.2"</span>, split.by <span class="op">=</span> <span class="st">"Prep.Method"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">markerp</span>, <span class="va">dp</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">6</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="integration_files/figure-html/unnamed-chunk-3-1.png" width="100%"></p>
<div class="section level3">
<h3 id="integrate-within-species">Integrate within species<a class="anchor" aria-label="anchor" href="#integrate-within-species"></a>
</h3>
<p>The <code><a href="../reference/integration_workflow.html">integration_workflow()</a></code> function can integrate
datasets obtained from mouse or human source. When integrating datasets
obtained from the same species, the data is processed according to
standard integration workflow.</p>
</div>
<div class="section level3">
<h3 id="integrate-accross-species">Integrate accross species<a class="anchor" aria-label="anchor" href="#integrate-accross-species"></a>
</h3>
<p><code><a href="../reference/integration_workflow.html">integration_workflow()</a></code> can also be used to integrate
datasets collected from different species, i.e., human and mouse. In
this case, the sub-function <code><a href="../reference/cross_species_integrate.html">cross_species_integrate()</a></code> first
converts the mouse seurat objects to human by transforming features in
the mouse dataset into human specific features and then integrates the
two objects using standard integration workflow.</p>
</div>
<div class="section level3">
<h3 id="steps-of-integration">Steps of integration<a class="anchor" aria-label="anchor" href="#steps-of-integration"></a>
</h3>
<p>The standard integration procedure in
<code><a href="../reference/integration_workflow.html">integration_workflow()</a></code> includes multiple sub-functions
responsible for performing different steps of integration.</p>
<p>Prior to integration, <code>seurat_preprocess</code> performs
standard preprocessing (log-normalization) and identifies variable
features individually for each element in the list. Then, anchors are
identified between the individual datasets using
<code>FindIntegrationAnchors</code> which is then used by
<code>IntegrateData</code>to return a Seurat object containing a new
Assay, which holds an integrated expression matrix for all cells.</p>
<p>Downstream analysis is performed on this integrated matrix by
<code>seurat_integration_pipeline</code> sub-function which scales the
integrated data, reduces the dimensions of seurat object, performs
clustering at different resolutions and identifies marker genes.</p>
<p>Finally, the resulting integrated datasets can be visualized by
plotting the output</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">integrated_seu</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, group.by <span class="op">=</span> <span class="st">"Prep.Method"</span>, <span class="op">)</span></span></code></pre></div>
<p><img src="integration_files/figure-html/unnamed-chunk-4-1.png" width="100%"></p>
</div>
</div>
<div class="section level2">
<h2 id="assumptions-of-the-input-data">Assumptions of the input data<a class="anchor" aria-label="anchor" href="#assumptions-of-the-input-data"></a>
</h2>
<p>Prior to integration, the Seurat object must be pre-processed using
the <code>object_preprocess</code> function. This function handles
normalization and scaling required for downstream data analysis.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Kevin Stachelek, Bhavana Bhat.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
