<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Shiny App • seuratTools</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Shiny App">
<meta property="og:description" content="seuratTools">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">seuratTools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.6.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/integration.html">Integration of Seurat objects</a>
    </li>
    <li>
      <a href="../articles/preprocessing.html">Preprocessing</a>
    </li>
    <li>
      <a href="../articles/shiny_app.html">Shiny App</a>
    </li>
    <li>
      <a href="../articles/visualization.html">Visualization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cobriniklab/seuratTools/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Shiny App</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/cobriniklab/seuratTools/blob/HEAD/vignettes/articles/shiny_app.Rmd" class="external-link"><code>vignettes/articles/shiny_app.Rmd</code></a></small>
      <div class="hidden name"><code>shiny_app.Rmd</code></div>

    </div>

    
    
<p>seuratTools allows exploratory data analysis of a single cell dataset
processed using <code>Seurat</code>. A demo using a human gene
transcript dataset from Shayler et al. (link) is available
<a href="https://docker.saban.chla.usc.edu/cobrinik/app/seuratApp/" target="_blank" rel="noopener noreferrer" class="external-link">here</a></p>
<p>Here we’ll provide a brief tour of this app organized by section with
app functions organized as tabs along the sidebar upon startup.</p>
<div class="section level3">
<h3 id="reformat-metadata">Reformat Metadata<a class="anchor" aria-label="anchor" href="#reformat-metadata"></a>
</h3>
<p>The metadata of the single cell dataset can be reformatted by
appending arbitrary variables to the cell metadata based on the results
of exploratory data analysis.</p>
<p>Metadata addition can be executed by uploading a csv with row names
matching cell ids with new variables as columns. Metadata editing can
also be accomplished using a built-in spreadsheet tool which displays
complete metadata values that can be edited by filtering and clicking
through the spreadsheet cells.</p>
</div>
<div class="section level3">
<h3 id="plot-data">Plot Data<a class="anchor" aria-label="anchor" href="#plot-data"></a>
</h3>
<div class="section level4">
<h4 id="dimensional-reduction-plots">Dimensional reduction plots<a class="anchor" aria-label="anchor" href="#dimensional-reduction-plots"></a>
</h4>
<p>seuratTools provides visualization of embeddings in PCA, tSNE, and
UMAP for scRNAseq data summarized at the gene and transcript level.
Plots can be customized to display cell metadata and the results of
community detection clustering by Louvain or Leiden algorithms as
implemented in Seurat. In addition, gene and transcript feature
expression can be overlaid on embedding plots both at the gene and
transcript level.</p>
<p><img src="images/louvain_and_feature_umap_plot.png" style="width:70.0%"></p>
</div>
<div class="section level4">
<h4 id="readumi-count-histograms">read/UMI count histograms<a class="anchor" aria-label="anchor" href="#readumi-count-histograms"></a>
</h4>
<p>To facilitate quality control, cell-level summarized umi and read
count values can also be overlaid with cell metadata and clustering
results.</p>
<p><img src="images/pair_of_histograms.png" style="width:70.0%"></p>
</div>
<div class="section level4">
<h4 id="clustering-trees">Clustering trees<a class="anchor" aria-label="anchor" href="#clustering-trees"></a>
</h4>
<p>A cluster tree of cell identities is displayed using <a href="https://lazappi.github.io/clustree/articles/clustree.html" class="external-link">scclustviz</a>.
This plot can be used to help establish an optimal clustering
resolution.</p>
<p><img src="images/clustree.png" style="width:70.0%"></p>
</div>
</div>
<div class="section level3">
<h3 id="heat-mapsviolin-plots">Heat Maps/Violin Plots<a class="anchor" aria-label="anchor" href="#heat-mapsviolin-plots"></a>
</h3>
<div class="section level4">
<h4 id="heat-maps">
<em>Heat Maps</em><a class="anchor" aria-label="anchor" href="#heat-maps"></a>
</h4>
<p>When plotting the heat map, gene/transcript expression values for
each cell are normalized by that cell’s total expression then multiplied
by 10,000 and natural-log transformed before plotting.</p>
<p>Clustering model or metadata for column arrangement is taken as a
parameter to produce a complex heat map.</p>
<p>By default, 50 most highly variable genes are displayed. However, an
arbitrary lists of genes can be plotted for comparison. The
genes/transcripts are displayed in the order they are listed.</p>
<p><img src="images/Heatmap2.0.png" style="width:70.0%"></p>
</div>
<div class="section level4">
<h4 id="violin-plots">
<em>Violin Plots</em><a class="anchor" aria-label="anchor" href="#violin-plots"></a>
</h4>
<p>Feature (gene or transcript) expression can be viewed by violin plot
based on sample metadata</p>
<p>Violin plots are a hybrid of density and box plots. It shows the
probability density of the data at different values providing a visual
representation of distribution of expression level of a feature in
different groups of cell. The horizontal line is a marker for the median
of the data and the box shows the interquartile ranges.</p>
<p>The parameters that could to be chosen here are: 1. ‘Grouping
variable’ 2. ‘Data Type (Transformed/Raw)’</p>
<p><img src="images/violinPlot.png" style="width:70.0%"></p>
</div>
</div>
<div class="section level3">
<h3 id="coverage-plots">Coverage plots<a class="anchor" aria-label="anchor" href="#coverage-plots"></a>
</h3>
<p>Fine-grained analysis of isoform contributions can be achieved by
plotting absolute read coverage plots across a given gene.</p>
<p>indicate the **read depths across all transcripts for different
groups of cells within a genomic region.</p>
<p>The three user input parameters are: 1. ‘Select a gene’- Select a
gene of interest 2. ‘Color by variable’- Select a variable by which to
group the cells by 3. ‘Groups to display’- Select the groups to be
displayed</p>
<p><img src="images/PlotCoverage2.0.png" style="width:70.0%"></p>
</div>
<div class="section level3">
<h3 id="differential-expression">Differential expression<a class="anchor" aria-label="anchor" href="#differential-expression"></a>
</h3>
<p>seuratTools provides different methods of differential expression
analysis that users can choose from to determine differential expression
of gene. DE methods included in seuratTools are t-test, Wilcoxon
rank-sum test, Likelihood ratio test, and MAST</p>
<p>Running DE testing results in a data frame containing the following
information: * p_val : Unadjusted p-value * avg_log2FC : log fold-change
of the average expression between the two groups. * pct.1 : The
percentage of cells where the feature is detected in the first group *
pct.2 : The percentage of cells where the feature is detected in the
second group * p_val_adj : Adjusted p-value, based on bonferroni
correction using all features in the dataset.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">d1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"Dominic.csv"</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="volcano-plot">Volcano Plot<a class="anchor" aria-label="anchor" href="#volcano-plot"></a>
</h4>
<p>The results of from the DE analysis is plotted using a volcano plot
that shows statistical significance (P value) versus magnitude of change
(fold change). Here, the cutoffs for log2FC and P value can be adjusted
to set thresholds for statistical significance using the contols
provided.</p>
<p><img src="images/VolcanoPlot.png" style="width:50.0%"></p>
</div>
</div>
<div class="section level3">
<h3 id="find-markers">Find Markers<a class="anchor" aria-label="anchor" href="#find-markers"></a>
</h3>
<p>Marker features based on louvain/leiden cluster identities or cell
metadata can be defined based on results of wilcoxon rank-sum test via
<a href="https://www.biorxiv.org/content/10.1101/653253v1" class="external-link">presto</a> or
measures of cluster specificity with genesorteR (Ibrahim, 2020) .
Variable numbers of marker genes per cell group can be specified based
on adjusted p value and thresholded log fold change.</p>
<p><img src="images/marker_gene_plot.png"></p>
</div>
<div class="section level3">
<h3 id="subset-seurat-input">Subset Seurat Input<a class="anchor" aria-label="anchor" href="#subset-seurat-input"></a>
</h3>
<p>It is often useful to subset a single cell data set based on cell
metadata whether experimentally determined (age, collection method,
etc.) or derived from analysis (quality control metrics, annotated cell
type)</p>
<p>seuratTools makes it simple to subset a dataset consisting of a
single batch or batch-integrated dataset. Subsetting can be accomplished
either in a graphical setting by lasso-selection from a dimensionally
reduced plot or by by specification of a formatted .csv file.</p>
<p>Subsetting of single or batch integrated data will trigger renewal of
all relevant preprocessing steps including dimensional reduction,
clustering, marker gene, and pathway enrichment as well as integration
based on a ‘batch’ variable</p>
</div>
<div class="section level3">
<h3 id="all-transcripts">All Transcripts<a class="anchor" aria-label="anchor" href="#all-transcripts"></a>
</h3>
<p>If expression is summarized with both gene and transcript assays, it
is possible to plot all constituent transcripts (labeled by ensembl
transcript ids) making up a given gene.</p>
<p>The two parameters that must be chosen are:</p>
<ol style="list-style-type: decimal">
<li>dimensional reduction method: PCA, UMAP or tSNE</li>
<li>The name of the gene of interest</li>
</ol>
<p><img src="images/unnamed-chunk-9-1.png" style="width:70.0%"></p>
<p>Direct comparison of the contribution of individual transcripts can
be achieved using stacked bar plots to answer the question: What
contribution does each transcript make for each gene expression?</p>
<p><img src="images/transcript_composisiont.png" style="width:70.0%"></p>
</div>
<div class="section level3">
<h3 id="pseudotime-analysis-by-monocle-3">Pseudotime Analysis by <a href="https://cole-trapnell-lab.github.io/monocle3/" class="external-link">Monocle 3</a><a class="anchor" aria-label="anchor" href="#pseudotime-analysis-by-monocle-3"></a>
</h3>
<p>Pseudotime analysis is broken into several steps, mirroring a typical
command-line workflow. 1. Users can subset an embedded dataset then
calculate a trajectory graph. 1. Root cells can be specified and cells
ordered in pseudotime with monocle3. 2. Features that vary significantly
over pseudotime based on autocorrelation can then be determined and
grouped into modules that vary similarly over pseudotime.</p>
<p><img src="images/monocle_first_steps.png" style="width:70.0%"></p>
<p>Heat maps and scatter plots are available at gene and module
levels.</p>
<p><img src="images/monocle_final_steps.png" style="width:70.0%"></p>
</div>
<div class="section level3">
<h3 id="regress-features">Regress Features<a class="anchor" aria-label="anchor" href="#regress-features"></a>
</h3>
<p>To correct for expression variation in analysis due to phenomena
unrelated to focus of the study, rather than excluding count values
attributable to such process, we can adjust the expression of all
remaining genes or transcripts in each cell based on the sum score of
relevant genes, to regress out cell-cycle effects, for <a href="https://satijalab.org/seurat/v3.0/cell_cycle_vignette.html" class="external-link">example</a></p>
<p>Such regression can be performed for any arbitrary gene set. Some
preset gene lists are included in <code>seuratTools</code> including
cell-cycle, mitochondrial, and apoptosis related gene sets.</p>
<p><img src="images/Screenshot_2023-08-25_at_1.40.35_PM.png" style="width:70.0%"></p>
</div>
<div class="section level3">
<h3 id="technical-information">Technical information<a class="anchor" aria-label="anchor" href="#technical-information"></a>
</h3>
<p>Technical information regarding upstream processing, software version
numbers, and dataset features can be viewed.</p>
<p><img src="images/Screenshot_2023-08-25_at_1.40.57_PM.png" style="width:70.0%"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Kevin Stachelek, Bhavana Bhat.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
