<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Visualization • seuFLViz</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Visualization">
<meta property="og:description" content="seuFLViz">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">seuFLViz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.6.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/integration.html">Integration of Seurat objects</a>
    </li>
    <li>
      <a href="../articles/pipeline.html">Pipeline</a>
    </li>
    <li>
      <a href="../articles/preprocessing.html">Preprocessing</a>
    </li>
    <li>
      <a href="../articles/shiny_app.html">Shiny App</a>
    </li>
    <li>
      <a href="../articles/visualization.html">Visualization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cobriniklab/seuFLViz/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Visualization</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/cobriniklab/seuFLViz/blob/HEAD/vignettes/articles/visualization.Rmd" class="external-link"><code>vignettes/articles/visualization.Rmd</code></a></small>
      <div class="hidden name"><code>visualization.Rmd</code></div>

    </div>

    
    
<p>This article demonstrates the visualization tools in seuFLViz.
We’ll introduce included functions, their usage, and resulting plots</p>
<p>First step is to load seuFLViz package and all other packages
required</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cobriniklab/seuFLViz" class="external-link">seuFLViz</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com" class="external-link">ggraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span></code></pre></div>
<p>seuFLViz contains many plotting functions that allows for
visualization, these plots can be customized for interactive or
non-interactive display.</p>
<div class="section level2">
<h2 id="plot-expression">Plot expression<a class="anchor" aria-label="anchor" href="#plot-expression"></a>
</h2>
<p>Expression of one or more features (genes or transcripts) can be
plotted on a given embedding, resulting in an interactive, single or
multi-feature plot.</p>
<p>When plotting only one feature, output is identical to
<code><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">Seurat::FeaturePlot</a></code></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_feature.html">plot_feature</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span>,</span>
<span>    embedding <span class="op">=</span> <span class="st">"umap"</span>,</span>
<span>    features <span class="op">=</span> <span class="st">"NRL"</span>, return_plotly <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-3-1.png" width="100%"></p>
<p>An interactive output plot can be generated by specifying
<code>return_plotly = TRUE</code> which uses <code>ggplotly</code>
allowing identification of individual cells for further
investigation.</p>
<div class="section level3">
<h3 id="plot-read-count-or-other-qc-measurements">Plot read count or other QC measurements<a class="anchor" aria-label="anchor" href="#plot-read-count-or-other-qc-measurements"></a>
</h3>
<p>The <code>plot_readcount</code> function displays a histogram of cell
read counts colored according to a categorical variable using the
argument <code>color.by</code>. Here we can see that read counts for
this dataset are not distinctly different depending on the method of
organoid preparation used (Kuwahara or Zhong)</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_readcount.html">plot_readcount</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span>,</span>
<span>    metavar <span class="op">=</span> <span class="st">"nCount_RNA"</span>,</span>
<span>    color.by <span class="op">=</span> <span class="st">"Prep.Method"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-4-1.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="plot">Plot<a class="anchor" aria-label="anchor" href="#plot"></a>
</h3>
<p>seuFLViz contains a function that produces an interactive scatter
plot of a metadata variable, where each point in the plot represents a
cell whose position on the plot is given by the cell embedding
determined by the dimensional reduction technique. The default dimension
reduction used is “UMAP”. The group argument in <code><a href="../reference/plot_var.html">plot_var()</a></code>
function allows the user to enter the metadata variable by which to
group the cells by, its default value is set to “batch”.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_var.html">plot_var</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span>, group <span class="op">=</span> <span class="st">"type"</span>, embedding <span class="op">=</span> <span class="st">"umap"</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-5-1.png" width="100%"></p>
<p>This function utilizes a Seurat function, <code><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot()</a></code>, as
sub function which produces the dimensional reduction plot.The
interactive parameter, <code>return_plotly</code>, in plot_var when set
to TRUE will convert the plot into an interactive plot using ggplotly
function from R’s plotly package</p>
</div>
<div class="section level3">
<h3 id="plot-cluster-marker-genes">Plot cluster marker genes<a class="anchor" aria-label="anchor" href="#plot-cluster-marker-genes"></a>
</h3>
<p>Marker genes of louvain clusters or additional experimental metadata
can be plotted using <code>plot_markers</code>. This allows
visualization of n marker features grouped by the metadata of interest.
Marker genes are identified using wilcoxon rank-sum test as implemented
in <code>presto</code>. In the resulting dot plot the size of the dot
corresponds to the percentage of cells expressing the feature in each
cluster and the color represents the average expression level of the
feature.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_markers.html">plot_markers</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span>,</span>
<span>    metavar <span class="op">=</span> <span class="st">"gene_snn_res.0.2"</span>,</span>
<span>    marker_method <span class="op">=</span> <span class="st">"presto"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-6-1.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="violin-plot">Violin plot<a class="anchor" aria-label="anchor" href="#violin-plot"></a>
</h3>
<p>To visualize the distribution of expression level of a feature in
different groups of cells seuFLViz draws a violin plot. This function
uses Seurat’s <code>VInPlot()</code> function as a sub-function to
create the violin plot, where the metadata variable, provided to the
function through the variable <code>plot_var</code>, is used to group
the cells and based on the level of feature expression a violin plot is
produced.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_violin.html">plot_violin</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span>, plot_var <span class="op">=</span> <span class="st">"type"</span>, features <span class="op">=</span> <span class="st">"NRL"</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-7-1.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="ploting-all-transcripts">Ploting all transcripts<a class="anchor" aria-label="anchor" href="#ploting-all-transcripts"></a>
</h3>
<p>This function plots the expression level of all the transcripts
present in a given Seurat object, coding for a feature/gene of interest.
The gene of interest is specified by the argument ‘features’ and by
default the data is superimposed on UMAP embedding.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_all_transcripts.html">plot_all_transcripts</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span>, features <span class="op">=</span> <span class="st">"NRL"</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-8-1.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="ploting-transcrpit-composition">Ploting transcrpit composition<a class="anchor" aria-label="anchor" href="#ploting-transcrpit-composition"></a>
</h3>
<p>This seuFLViz function produces a visual representation of the
expression levels of each transcript in a gene map. The function
plot_transcript_composition() plots the proportion of reads of a given
gene map to each transcript. The gene of interest is specified by the
argument ‘gene_symbol’.</p>
<p><img src="visualization_files/figure-html/unnamed-chunk-9-1.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="ploting-ridge-plots-for-cell-cycle-scoring">Ploting ridge plots for cell cycle scoring<a class="anchor" aria-label="anchor" href="#ploting-ridge-plots-for-cell-cycle-scoring"></a>
</h3>
<p>The plot_ridge() function in seuFLViz plots a ridge plot for cell
cycle scoring of a feature, providing a visual representation of
expression level of the feature in each cell cycle. This function uses
two embedded Seurat functions. The function CellCycleScoring() assigns
each cell a score based on its expression of G2/M and S phase markers.
This data is then used by the function, RidgePlot(), which uses user
provided ‘features’ argument to plot the ridge plot of the features.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_cell_cycle_distribution.html">plot_cell_cycle_distribution</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span>, <span class="st">"NRL"</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-10-1.png" width="100%"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/plot_cell_cycle_distribution.html">plot_cell_cycle_distribution</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span>, <span class="st">"nFeature_RNA"</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-10-2.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="plot-complex-heatmap-from-seurat-object">Plot Complex heatmap from Seurat object<a class="anchor" aria-label="anchor" href="#plot-complex-heatmap-from-seurat-object"></a>
</h3>
<p>The seuFLViz function plots an annotated complex heat map from the
Seurat object with a dendogram added to the left side and the top.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_10_features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="../reference/seu_complex_heatmap.html">seu_complex_heatmap</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span>,</span>
<span>    features <span class="op">=</span> <span class="va">top_10_features</span>,</span>
<span>    group.by <span class="op">=</span> <span class="st">"gene_snn_res.0.2"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-11-1.png" width="100%"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/seu_complex_heatmap.html">seu_complex_heatmap</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span>,</span>
<span>    features <span class="op">=</span> <span class="va">top_10_features</span>,</span>
<span>    group.by <span class="op">=</span> <span class="st">"gene_snn_res.0.2"</span>, col_arrangement <span class="op">=</span> <span class="st">"average"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-11-2.png" width="100%"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/seu_complex_heatmap.html">seu_complex_heatmap</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span>,</span>
<span>    features <span class="op">=</span> <span class="va">top_10_features</span>,</span>
<span>    group.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Prep.Method"</span>, <span class="st">"gene_snn_res.0.2"</span><span class="op">)</span>,</span>
<span>    col_arrangement <span class="op">=</span> <span class="st">"gene_snn_res.0.2"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-11-3.png" width="100%"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/seu_complex_heatmap.html">seu_complex_heatmap</a></span><span class="op">(</span><span class="va">human_gene_transcript_seu</span>,</span>
<span>    features <span class="op">=</span> <span class="va">top_10_features</span>,</span>
<span>    group.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Prep.Method"</span>, <span class="st">"gene_snn_res.0.2"</span><span class="op">)</span>,</span>
<span>    col_arrangement <span class="op">=</span> <span class="st">"Prep.Method"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-11-4.png" width="100%"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Kevin Stachelek, Bhavana Bhat.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
