<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ seuratTools</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">seuratTools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.6.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/integration.html">Integration of Seurat objects</a>
    </li>
    <li>
      <a href="../articles/preprocessing.html">Preprocessing</a>
    </li>
    <li>
      <a href="../articles/shiny_app.html">Shiny App</a>
    </li>
    <li>
      <a href="../articles/visualization.html">Visualization</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/cobriniklab/seuratTools/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="Monocle2_diffex.html">Monocle2_diffex()</a></code> </p>
        </td>
        <td><p>Monocle2 differential expression</p></td>
      </tr><tr><td>
          <p><code><a href="SetDefaultAssay.html">SetDefaultAssay()</a></code> </p>
        </td>
        <td><p>Reset default assay</p></td>
      </tr><tr><td>
          <p><code><a href="add_census_slot.html">add_census_slot()</a></code> </p>
        </td>
        <td><p>add census assay to a seurat object</p></td>
      </tr><tr><td>
          <p><code><a href="add_percent_mito.html">add_percent_mito()</a></code> </p>
        </td>
        <td><p>Annotate percent mitochondrial reads per cell</p></td>
      </tr><tr><td>
          <p><code><a href="add_read_count_col.html">add_read_count_col()</a></code> </p>
        </td>
        <td><p>Annotate Low Read Count Category</p></td>
      </tr><tr><td>
          <p><code><a href="allTranscripts.html">allTranscripts()</a></code> </p>
        </td>
        <td><p>Plot All Transcripts Server</p></td>
      </tr><tr><td>
          <p><code><a href="allTranscriptsui.html">allTranscriptsui()</a></code> </p>
        </td>
        <td><p>Plot All Transcripts UI Module</p></td>
      </tr><tr><td>
          <p><code><a href="annotate_cell_cycle.html">annotate_cell_cycle()</a></code> </p>
        </td>
        <td><p>Annotate Cell Cycle</p></td>
      </tr><tr><td>
          <p><code><a href="annotate_excluded.html">annotate_excluded()</a></code> </p>
        </td>
        <td><p>Annotate Exclusion Criteria</p></td>
      </tr><tr><td>
          <p><code><a href="append_to_project_db.html">append_to_project_db()</a></code> </p>
        </td>
        <td><p>Update a database of seuratTools projects</p></td>
      </tr><tr><td>
          <p><code><a href="arrange_ptime_heatmaps.html">arrange_ptime_heatmaps()</a></code> </p>
        </td>
        <td><p>Arrange processed monocle CDS objects containing heatmaps</p></td>
      </tr><tr><td>
          <p><code><a href="assign_clusters_to_cds.html">assign_clusters_to_cds()</a></code> </p>
        </td>
        <td><p>Assign Clusters to CDS</p></td>
      </tr><tr><td>
          <p><code><a href="build_bigwig_db.html">build_bigwig_db()</a></code> </p>
        </td>
        <td><p>Create a database of bigwigfiles</p></td>
      </tr><tr><td>
          <p><code><a href="by_row_corr_between_pt_heatmaps.html">by_row_corr_between_pt_heatmaps()</a></code> </p>
        </td>
        <td><p>calculate row-wise correlation between two dataframes</p></td>
      </tr><tr><td>
          <p><code><a href="calc_cor_across_heatmaps.html">calc_cor_across_heatmaps()</a></code> </p>
        </td>
        <td><p>calculate correlation between heatmap matrices in processed cds_gene_list</p></td>
      </tr><tr><td>
          <p><code><a href="calc_pseudotime_heatmap.html">calc_pseudotime_heatmap()</a></code> </p>
        </td>
        <td><p>calculate pseudotime heatmaps</p></td>
      </tr><tr><td>
          <p><code><a href="ccScore.html">ccScore()</a></code> </p>
        </td>
        <td><p>Cell Cycle Score</p></td>
      </tr><tr><td>
          <p><code><a href="ccScoreui.html">ccScoreui()</a></code> </p>
        </td>
        <td><p>Cell Cycle Score UI</p></td>
      </tr><tr><td>
          <p><code><a href="cells_selected.html">cells_selected()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr><td>
          <p><code><a href="celltype_markers.html">celltype_markers</a></code> </p>
        </td>
        <td><p>Retinal Cell Type Marker Genes</p></td>
      </tr><tr><td>
          <p><code><a href="changeEmbedParams.html">changeEmbedParams()</a></code> </p>
        </td>
        <td><p>Change Embedding Parameters</p></td>
      </tr><tr><td>
          <p><code><a href="changeEmbedParamsui.html">changeEmbedParamsui()</a></code> </p>
        </td>
        <td><p>Change Embedding Parameters UI</p></td>
      </tr><tr><td>
          <p><code><a href="clustering_workflow.html">clustering_workflow()</a></code> </p>
        </td>
        <td><p>Clustering Workflow</p></td>
      </tr><tr><td>
          <p><code><a href="combine_cols.html">combine_cols()</a></code> </p>
        </td>
        <td><p>Reformat Seurat Object Metadata</p></td>
      </tr><tr><td>
          <p><code><a href="combine_looms.html">combine_looms()</a></code> </p>
        </td>
        <td><p>Combine Loom Files</p></td>
      </tr><tr><td>
          <p><code><a href="convert_human_seu_to_mouse.html">convert_human_seu_to_mouse()</a></code> </p>
        </td>
        <td><p>Convert Seurat Objects from Human to Mouse</p></td>
      </tr><tr><td>
          <p><code><a href="convert_monoclev2_to_seuv3.html">convert_monoclev2_to_seuv3()</a></code> </p>
        </td>
        <td><p>Convert a Seurat V3 object to a Monocle v2 object</p></td>
      </tr><tr><td>
          <p><code><a href="convert_mouse_seu_to_human.html">convert_mouse_seu_to_human()</a></code> </p>
        </td>
        <td><p>Convert Seurat Objects from Mouse to Human</p></td>
      </tr><tr><td>
          <p><code><a href="convert_seu_list_to_multimodal.html">convert_seu_list_to_multimodal()</a></code> </p>
        </td>
        <td><p>convert seurat list to multimodal object</p></td>
      </tr><tr><td>
          <p><code><a href="convert_seu_to_cds.html">convert_seu_to_cds()</a></code> </p>
        </td>
        <td><p>Convert a Seurat Object to a Monocle Cell Data Set</p></td>
      </tr><tr><td>
          <p><code><a href="convert_seurat_to_sce.html">convert_seurat_to_sce()</a></code> </p>
        </td>
        <td><p>convert seurat object to cds</p></td>
      </tr><tr><td>
          <p><code><a href="convert_seuv3_to_monoclev2.html">convert_seuv3_to_monoclev2()</a></code> </p>
        </td>
        <td><p>Convert a Seurat V3 object to a Monocle v2 object</p></td>
      </tr><tr><td>
          <p><code><a href="convert_symbols_by_species.html">convert_symbols_by_species()</a></code> </p>
        </td>
        <td><p>Convert gene symbols between mouse and human</p></td>
      </tr><tr><td>
          <p><code><a href="convert_v3_to_v5.html">convert_v3_to_v5()</a></code> </p>
        </td>
        <td><p>Convert seurat object to seurat V5 format</p></td>
      </tr><tr><td>
          <p><code><a href="create_proj_matrix.html">create_proj_matrix()</a></code> </p>
        </td>
        <td><p>Create a Table of single Cell Projects</p></td>
      </tr><tr><td>
          <p><code><a href="create_project_db.html">create_project_db()</a></code> </p>
        </td>
        <td><p>Create a database of seuratTools projects</p></td>
      </tr><tr><td>
          <p><code><a href="cross_check_heatmaps.html">cross_check_heatmaps()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr><td>
          <p><code><a href="cross_species_integrate.html">cross_species_integrate()</a></code> </p>
        </td>
        <td><p>Integrate Seurat Objects from Mouse to Human</p></td>
      </tr><tr><td>
          <p><code><a href="default_helper.html">default_helper()</a></code> </p>
        </td>
        <td><p>Default Shiny Helper</p></td>
      </tr><tr><td>
          <p><code><a href="diffex.html">diffex()</a></code> </p>
        </td>
        <td><p>Differential Expression</p></td>
      </tr><tr><td>
          <p><code><a href="diffexui.html">diffexui()</a></code> </p>
        </td>
        <td><p>Differential Expression UI</p></td>
      </tr><tr><td>
          <p><code><a href="dockerSeuratApp.html">dockerSeuratApp()</a></code> </p>
        </td>
        <td><p>Create a seurat app on a shinyproxy docker instance</p></td>
      </tr><tr><td>
          <p><code><a href="dot-send_enrichr_query.html">.send_enrichr_query()</a></code> </p>
        </td>
        <td><p>Gene enrichment using Enrichr.</p></td>
      </tr><tr><td>
          <p><code><a href="dropdownButton.html">dropdownButton()</a></code> </p>
        </td>
        <td><p>Custom Dropdown Button</p></td>
      </tr><tr><td>
          <p><code><a href="enframe_markers.html">enframe_markers()</a></code> </p>
        </td>
        <td><p>Enframe seurat markers</p></td>
      </tr><tr><td>
          <p><code><a href="filter_low_rc_cells.html">filter_low_rc_cells()</a></code> </p>
        </td>
        <td><p>Filter our Cells from Seurat below read count threshold</p></td>
      </tr><tr><td>
          <p><code><a href="filter_merged_seu.html">filter_merged_seu()</a></code> </p>
        </td>
        <td><p>Filter a Single Seurat Object</p></td>
      </tr><tr><td>
          <p><code><a href="filter_merged_seus.html">filter_merged_seus()</a></code> </p>
        </td>
        <td><p>Filter a List of Seurat Objects</p></td>
      </tr><tr><td>
          <p><code><a href="filter_rows_to_top.html">filter_rows_to_top()</a></code> </p>
        </td>
        <td><p>Filter Rows to Top</p></td>
      </tr><tr><td>
          <p><code><a href="findMarkers.html">findMarkers()</a></code> </p>
        </td>
        <td><p>Find Markers</p></td>
      </tr><tr><td>
          <p><code><a href="findMarkersui.html">findMarkersui()</a></code> </p>
        </td>
        <td><p>Find Markers UI</p></td>
      </tr><tr><td>
          <p><code><a href="find_all_markers.html">find_all_markers()</a></code> </p>
        </td>
        <td><p>Find All Markers</p></td>
      </tr><tr><td>
          <p><code><a href="flip_pseudotime.html">flip_pseudotime()</a></code> </p>
        </td>
        <td><p>Flip Pseudotime</p></td>
      </tr><tr><td>
          <p><code><a href="format_new_metadata.html">format_new_metadata()</a></code> </p>
        </td>
        <td><p>Add Seurat Object Metadata</p></td>
      </tr><tr><td>
          <p><code><a href="format_pathway_table.html">format_pathway_table()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr><td>
          <p><code><a href="gene_test_plot.html">gene_test_plot()</a></code> </p>
        </td>
        <td><p>Plot Gene expression over pseudotime on reference and query trajectories from cellAlign</p></td>
      </tr><tr><td>
          <p><code><a href="genes_to_transcripts.html">genes_to_transcripts()</a></code> </p>
        </td>
        <td><p>Gene Symbols to Ensembl Transcript Ids</p></td>
      </tr><tr><td>
          <p><code><a href="getEnrichedPathways.html">getEnrichedPathways()</a></code> </p>
        </td>
        <td><p>Get enriched pathways based on marker genes from EnrichR.</p></td>
      </tr><tr><td>
          <p><code><a href="get_meta.html">get_meta()</a></code> </p>
        </td>
        <td><p>Get Metadata for a project</p></td>
      </tr><tr><td>
          <p><code><a href="get_transcripts_from_seu.html">get_transcripts_from_seu()</a></code> </p>
        </td>
        <td><p>Get Transcripts in Seurat Object</p></td>
      </tr><tr><td>
          <p><code><a href="global_cellalign.html">global_cellalign()</a></code> </p>
        </td>
        <td><p>Global Alignment of two trajectories with cellAlign</p></td>
      </tr><tr><td>
          <p><code><a href="harmony_integrate.html">harmony_integrate()</a></code> </p>
        </td>
        <td><p>Integrate small datasets with harmony</p></td>
      </tr><tr><td>
          <p><code><a href="human_to_mouse_homologs.html">human_to_mouse_homologs</a></code> </p>
        </td>
        <td><p>Gene Homologs Between Human and Mouse</p></td>
      </tr><tr><td>
          <p><code><a href="integrateProj.html">integrateProj()</a></code> </p>
        </td>
        <td><p>Integrate Projects Server Function</p></td>
      </tr><tr><td>
          <p><code><a href="integrateProjui.html">integrateProjui()</a></code> </p>
        </td>
        <td><p>Integrate Project UI</p></td>
      </tr><tr><td>
          <p><code><a href="integration_workflow.html">integration_workflow()</a></code> </p>
        </td>
        <td><p>Integration Workflow</p></td>
      </tr><tr><td>
          <p><code><a href="label_transfer.html">label_transfer()</a></code> </p>
        </td>
        <td><p>Transfer Labels Between Seurat Objects</p></td>
      </tr><tr><td>
          <p><code><a href="learn_graph_by_resolution.html">learn_graph_by_resolution()</a></code> </p>
        </td>
        <td><p>Learn Monocle Graph by Resolution</p></td>
      </tr><tr><td>
          <p><code><a href="list_plot_types.html">list_plot_types()</a></code> </p>
        </td>
        <td><p>Collate list of variables to be plotted</p></td>
      </tr><tr><td>
          <p><code><a href="load_bigwigs.html">load_bigwigs()</a></code> </p>
        </td>
        <td><p>Load Bigwigs</p></td>
      </tr><tr><td>
          <p><code><a href="load_counts_by_tximport.html">load_counts_by_tximport()</a></code> </p>
        </td>
        <td><p>Run <a href="http://bioconductor.org/packages/release/bioc/html/tximport.html" class="external-link">tximport</a> on a set of cells</p></td>
      </tr><tr><td>
          <p><code><a href="load_meta.html">load_meta()</a></code> </p>
        </td>
        <td><p>Load Sample Metadata for a given project</p></td>
      </tr><tr><td>
          <p><code><a href="load_seurat_from_proj.html">load_seurat_from_proj()</a></code> </p>
        </td>
        <td><p>Load Seurat Files from a single project path</p></td>
      </tr><tr><td>
          <p><code><a href="load_seurat_path.html">load_seurat_path()</a></code> </p>
        </td>
        <td><p>Read in Gene and Transcript Seurat Objects</p></td>
      </tr><tr><td>
          <p><code><a href="local_cellalign.html">local_cellalign()</a></code> </p>
        </td>
        <td><p>Align two trajectories via cellalign</p></td>
      </tr><tr><td>
          <p><code><a href="make_bigwig_db.html">make_bigwig_db()</a></code> </p>
        </td>
        <td><p>Make Bigwig Database</p></td>
      </tr><tr><td>
          <p><code><a href="make_seuratTools_clean_names.html">make_seuratTools_clean_names()</a></code> </p>
        </td>
        <td><p>Clean Vector of seuratTools Names</p></td>
      </tr><tr><td>
          <p><code><a href="mem_str.html">mem_str()</a></code> </p>
        </td>
        <td><p>Profile the memory of an s4object</p></td>
      </tr><tr><td>
          <p><code><a href="merge_loom.html">merge_loom()</a></code> </p>
        </td>
        <td><p>scvelo_assay</p></td>
      </tr><tr><td>
          <p><code><a href="merge_small_seus.html">merge_small_seus()</a></code> </p>
        </td>
        <td><p>Merge Small Seurat Objects</p></td>
      </tr><tr><td>
          <p><code><a href="metadata_from_batch.html">metadata_from_batch()</a></code> </p>
        </td>
        <td><p>Retrieve Metadata from Batch</p></td>
      </tr><tr><td>
          <p><code><a href="minimalSeuratApp.html">minimalSeuratApp()</a></code> </p>
        </td>
        <td><p>Create a minimal seurat app</p></td>
      </tr><tr><td>
          <p><code><a href="monocle.html">monocle()</a></code> </p>
        </td>
        <td><p>Monocle Server Module</p></td>
      </tr><tr><td>
          <p><code><a href="monocle_module_heatmap.html">monocle_module_heatmap()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr><td>
          <p><code><a href="monocleui.html">monocleui()</a></code> </p>
        </td>
        <td><p>Monocle UI Module</p></td>
      </tr><tr><td>
          <p><code><a href="pad_sample_files.html">pad_sample_files()</a></code> </p>
        </td>
        <td><p>pad sample numbers to prep for armor</p></td>
      </tr><tr><td>
          <p><code><a href="pathwayEnrichment.html">pathwayEnrichment()</a></code> </p>
        </td>
        <td><p>pathway enrichment</p></td>
      </tr><tr><td>
          <p><code><a href="pathwayEnrichmentui.html">pathwayEnrichmentui()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr><td>
          <p><code><a href="plotClustree.html">plotClustree()</a></code> </p>
        </td>
        <td><p>plot clustree server</p></td>
      </tr><tr><td>
          <p><code><a href="plotClustree_UI.html">plotClustree_UI()</a></code> </p>
        </td>
        <td><p>plot clustree ui</p></td>
      </tr><tr><td>
          <p><code><a href="plotCoverage.html">plotCoverage()</a></code> </p>
        </td>
        <td><p>Plot Coverage Module</p></td>
      </tr><tr><td>
          <p><code><a href="plotCoverage_UI.html">plotCoverage_UI()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr><td>
          <p><code><a href="plotDimRed.html">plotDimRed()</a></code> </p>
        </td>
        <td><p>Plot Dimensional Reduduction</p></td>
      </tr><tr><td>
          <p><code><a href="plotDimRedui.html">plotDimRedui()</a></code> </p>
        </td>
        <td><p>Plot Dimensional Reduduction UI</p></td>
      </tr><tr><td>
          <p><code><a href="plotHeatmap.html">plotHeatmap()</a></code> </p>
        </td>
        <td><p>Plot Heatmap</p></td>
      </tr><tr><td>
          <p><code><a href="plotHeatmapui.html">plotHeatmapui()</a></code> </p>
        </td>
        <td><p>Plot Heatmap ui</p></td>
      </tr><tr><td>
          <p><code><a href="plotReadCount.html">plotReadCount()</a></code> </p>
        </td>
        <td><p>Plot Read Count</p></td>
      </tr><tr><td>
          <p><code><a href="plotReadCountui.html">plotReadCountui()</a></code> </p>
        </td>
        <td><p>Plot Read Count UI</p></td>
      </tr><tr><td>
          <p><code><a href="plotViolin.html">plotViolin()</a></code> </p>
        </td>
        <td><p>Plot Violin Server</p></td>
      </tr><tr><td>
          <p><code><a href="plotViolinui.html">plotViolinui()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr><td>
          <p><code><a href="plot_all_ptimes.html">plot_all_ptimes()</a></code> </p>
        </td>
        <td><p>Plot a Set of Heatmaps Based on a provided pseudotime</p></td>
      </tr><tr><td>
          <p><code><a href="plot_all_transcripts.html">plot_all_transcripts()</a></code> </p>
        </td>
        <td><p>Plot All Transcripts</p></td>
      </tr><tr><td>
          <p><code><a href="plot_cds.html">plot_cds()</a></code> </p>
        </td>
        <td><p>Plot a Monocle Cell Data Set</p></td>
      </tr><tr><td>
          <p><code><a href="plot_cell_cycle_distribution.html">plot_cell_cycle_distribution()</a></code> </p>
        </td>
        <td><p>Plot cell cycle distribution grouped by metadata</p></td>
      </tr><tr><td>
          <p><code><a href="plot_cells.html">plot_cells()</a></code> </p>
        </td>
        <td><p>Plot cells of a monocle cell data set</p></td>
      </tr><tr><td>
          <p><code><a href="plot_feature.html">plot_feature()</a></code> </p>
        </td>
        <td><p>Plot Feature</p></td>
      </tr><tr><td>
          <p><code><a href="plot_feature_in_ref_query_ptime.html">plot_feature_in_ref_query_ptime()</a></code> </p>
        </td>
        <td><p>Plot Expression of a Given Feature of a set of Pseudotimes</p></td>
      </tr><tr><td>
          <p><code><a href="plot_gene_coverage_by_var.html">plot_gene_coverage_by_var()</a></code> </p>
        </td>
        <td><p>Plot BigWig Coverage for Genes of Interest by a Given Variable</p></td>
      </tr><tr><td>
          <p><code><a href="plot_markers.html">plot_markers()</a></code> </p>
        </td>
        <td><p>Plot Cluster Marker Genes</p></td>
      </tr><tr><td>
          <p><code><a href="plot_monocle_features.html">plot_monocle_features()</a></code> </p>
        </td>
        <td><p>Plot feature expression on a Monocle Cell Data Set</p></td>
      </tr><tr><td>
          <p><code><a href="plot_multiple_branches_heatmap.html">plot_multiple_branches_heatmap()</a></code> </p>
        </td>
        <td><p>Plot monocle pseudotime over multiple branches</p></td>
      </tr><tr><td>
          <p><code><a href="plot_pseudotime.html">plot_pseudotime()</a></code> </p>
        </td>
        <td><p>Plot pseudotime on a Monocle Cell Data Set</p></td>
      </tr><tr><td>
          <p><code><a href="plot_pseudotime_heatmap.html">plot_pseudotime_heatmap()</a></code> </p>
        </td>
        <td><p>plot pseudotime heatmap</p></td>
      </tr><tr><td>
          <p><code><a href="plot_readcount.html">plot_readcount()</a></code> </p>
        </td>
        <td><p>Plot Read Count</p></td>
      </tr><tr><td>
          <p><code><a href="plot_scvelo.html">plot_scvelo()</a></code> </p>
        </td>
        <td><p>plot scvelo</p></td>
      </tr><tr><td>
          <p><code><a href="plot_transcript_composition.html">plot_transcript_composition()</a></code> </p>
        </td>
        <td><p>Plot Transcript Composition</p></td>
      </tr><tr><td>
          <p><code><a href="plot_var.html">plot_var()</a></code> </p>
        </td>
        <td><p>Plot Metadata Variables</p></td>
      </tr><tr><td>
          <p><code><a href="plot_violin.html">plot_violin()</a></code> </p>
        </td>
        <td><p>Plot Violin plot</p></td>
      </tr><tr><td>
          <p><code><a href="plotly_settings.html">plotly_settings()</a></code> </p>
        </td>
        <td><p>Plotly settings</p></td>
      </tr><tr><td>
          <p><code><a href="prep_armor_meta.html">prep_armor_meta()</a></code> </p>
        </td>
        <td><p>prep armor metadata file</p></td>
      </tr><tr><td>
          <p><code><a href="prep_plot_genes_in_pseudotime.html">prep_plot_genes_in_pseudotime()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr><td>
          <p><code><a href="prep_slider_values.html">prep_slider_values()</a></code> </p>
        </td>
        <td><p>Prep Slider Values</p></td>
      </tr><tr><td>
          <p><code><a href="process_monocle_child.html">process_monocle_child()</a></code> </p>
        </td>
        <td><p>Preprocess a Monocle v2 object for heatmap based on provided pseudotime</p></td>
      </tr><tr><td>
          <p><code><a href="propagate_spreadsheet_changes.html">propagate_spreadsheet_changes()</a></code> </p>
        </td>
        <td><p>Propagate Metadata Changes</p></td>
      </tr><tr><td>
          <p><code><a href="query_experiment.html">query_experiment()</a></code> </p>
        </td>
        <td><p>Query Assay</p></td>
      </tr><tr><td>
          <p><code><a href="read_project_db.html">read_project_db()</a></code> </p>
        </td>
        <td><p>Read a database of seuratTools projects</p></td>
      </tr><tr><td>
          <p><code><a href="record_experiment_data.html">record_experiment_data()</a></code> </p>
        </td>
        <td><p>Record Experiment Metadata</p></td>
      </tr><tr><td>
          <p><code><a href="reference_integrate.html">reference_integrate()</a></code> </p>
        </td>
        <td><p>Transfer labels between gene or transcript objects</p></td>
      </tr><tr><td>
          <p><code><a href="reformatMetadataDR.html">reformatMetadataDR()</a></code> </p>
        </td>
        <td><p>Reformat Seurat Object Metadata Server</p></td>
      </tr><tr><td>
          <p><code><a href="reformatMetadataDRui.html">reformatMetadataDRui()</a></code> </p>
        </td>
        <td><p>Reformat Seurat Object Metadata UI</p></td>
      </tr><tr><td>
          <p><code><a href="regress_by_features.html">regress_by_features()</a></code> </p>
        </td>
        <td><p>Regress Seurat Object by Given Set of Genes</p></td>
      </tr><tr><td>
          <p><code><a href="reintegrate_seu.html">reintegrate_seu()</a></code> </p>
        </td>
        <td><p>Reintegrate (filtered) seurat objects</p></td>
      </tr><tr><td>
          <p><code><a href="rename_from_x_notation.html">rename_from_x_notation()</a></code> </p>
        </td>
        <td><p>Rename cell ids from annoying old notation</p></td>
      </tr><tr><td>
          <p><code><a href="rename_seurat.html">rename_seurat()</a></code> </p>
        </td>
        <td><p>Give a new project name to a seurat object</p></td>
      </tr><tr><td>
          <p><code><a href="reorg_seurat_files.html">reorg_seurat_files()</a></code> </p>
        </td>
        <td><p>Reorganize seurat objects to a multimodal format</p></td>
      </tr><tr><td>
          <p><code><a href="reset_seu_meta.html">reset_seu_meta()</a></code> </p>
        </td>
        <td><p>Reset Seurat Metadata</p></td>
      </tr><tr><td>
          <p><code><a href="run_BEAM.html">run_BEAM()</a></code> </p>
        </td>
        <td><p>Run BEAM from Monocle2</p></td>
      </tr><tr><td>
          <p><code><a href="run_census.html">run_census()</a></code> </p>
        </td>
        <td><p>Run Census</p></td>
      </tr><tr><td>
          <p><code><a href="run_enrichmentbrowser.html">run_enrichmentbrowser()</a></code> </p>
        </td>
        <td><p>Run Enrichment Browser on Differentially Expressed Genes</p></td>
      </tr><tr><td>
          <p><code><a href="run_hmap.html">run_hmap()</a></code> </p>
        </td>
        <td><p>Run heatmap from Monocle2</p></td>
      </tr><tr><td>
          <p><code><a href="run_seurat_de.html">run_seurat_de()</a></code> </p>
        </td>
        <td><p>Run Seurat Differential Expression</p></td>
      </tr><tr><td>
          <p><code><a href="save_seurat.html">save_seurat()</a></code> </p>
        </td>
        <td><p>Save seurat object to <project>/output/sce/<feature>_seu.rds</feature></project></p></td>
      </tr><tr><td>
          <p><code><a href="sce_from_tibbles.html">sce_from_tibbles()</a></code> </p>
        </td>
        <td><p>Create Single Cell Experiment from Tibbles</p></td>
      </tr><tr><td>
          <p><code><a href="scvelo_expression.html">scvelo_expression()</a></code> </p>
        </td>
        <td><p>plot  scvelo expression</p></td>
      </tr><tr><td>
          <p><code><a href="set_colnames_txi.html">set_colnames_txi()</a></code> </p>
        </td>
        <td><p>Set Column Names from <code>tximport</code></p></td>
      </tr><tr><td>
          <p><code><a href="seu_calcn.html">seu_calcn()</a></code> </p>
        </td>
        <td><p>Calculate Read Count Metrics for a Seurat object</p></td>
      </tr><tr><td>
          <p><code><a href="seu_complex_heatmap.html">seu_complex_heatmap()</a></code> </p>
        </td>
        <td><p>Plot Annotated Complexheatmap from Seurat object</p></td>
      </tr><tr><td>
          <p><code><a href="seu_from_tibbles.html">seu_from_tibbles()</a></code> </p>
        </td>
        <td><p>Create a Seurat Object from a set of tibbles</p></td>
      </tr><tr><td>
          <p><code><a href="seu_from_tximport.html">seu_from_tximport()</a></code> </p>
        </td>
        <td><p>Create a seurat object from output of  <a href="http://bioconductor.org/packages/release/bioc/html/tximport.html" class="external-link">tximport</a> and a table of cell metadata</p></td>
      </tr><tr><td>
          <p><code><a href="seuratApp.html">seuratApp()</a></code> </p>
        </td>
        <td><p>Create a shiny app for a project on disk</p></td>
      </tr><tr><td>
          <p><code><a href="seuratToolsBox.html">seuratToolsBox()</a></code> </p>
        </td>
        <td><p>custom collapsible box</p></td>
      </tr><tr><td>
          <p><code><a href="seurat_cluster.html">seurat_cluster()</a></code> </p>
        </td>
        <td><p>Run Louvain Clustering at Multiple Resolutions</p></td>
      </tr><tr><td>
          <p><code><a href="seurat_find_markers.html">seurat_find_markers()</a></code> </p>
        </td>
        <td><p>Find Cell Type Markers in a Seurat Object</p></td>
      </tr><tr><td>
          <p><code><a href="seurat_integrate.html">seurat_integrate()</a></code> </p>
        </td>
        <td><p>Batch Correct Multiple Seurat Objects</p></td>
      </tr><tr><td>
          <p><code><a href="seurat_integration_pipeline.html">seurat_integration_pipeline()</a></code> </p>
        </td>
        <td><p>Run Seurat Integration</p></td>
      </tr><tr><td>
          <p><code><a href="seurat_pipeline.html">seurat_pipeline()</a></code> </p>
        </td>
        <td><p>Run Seurat Pipeline</p></td>
      </tr><tr><td>
          <p><code><a href="seurat_preprocess.html">seurat_preprocess()</a></code> </p>
        </td>
        <td><p>Preprocess Seurat Object</p></td>
      </tr><tr><td>
          <p><code><a href="seurat_reduce_dimensions.html">seurat_reduce_dimensions()</a></code> </p>
        </td>
        <td><p>Dimensional Reduction</p></td>
      </tr><tr><td>
          <p><code><a href="seurat_v5_integrate.html">seurat_v5_integrate()</a></code> </p>
        </td>
        <td><p>Batch Correct Multiple Seurat Objects</p></td>
      </tr><tr><td>
          <p><code><a href="seurat_v5_integration_pipeline.html">seurat_v5_integration_pipeline()</a></code> </p>
        </td>
        <td><p>Run Seurat Integration</p></td>
      </tr><tr><td>
          <p><code><a href="stash_marker_features.html">stash_marker_features()</a></code> </p>
        </td>
        <td><p>Stash Marker Genes in a Seurat Object</p></td>
      </tr><tr><td>
          <p><code><a href="subset_by_meta.html">subset_by_meta()</a></code> </p>
        </td>
        <td><p>Subset by new metadata</p></td>
      </tr><tr><td>
          <p><code><a href="swap_counts_from_feature.html">swap_counts_from_feature()</a></code> </p>
        </td>
        <td><p>Swap counts from Feature</p></td>
      </tr><tr><td>
          <p><code><a href="tableSelected.html">tableSelected()</a></code> </p>
        </td>
        <td><p>Create Table of Selected Cells</p></td>
      </tr><tr><td>
          <p><code><a href="tableSelectedui.html">tableSelectedui()</a></code> </p>
        </td>
        <td><p>Create Table of Selected Cells UI</p></td>
      </tr><tr><td>
          <p><code><a href="techInfo.html">techInfo()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr><td>
          <p><code><a href="techInfoui.html">techInfoui()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr><td>
          <p><code><a href="threshold_monocle_genes.html">threshold_monocle_genes()</a></code> </p>
        </td>
        <td><p>Threshold monocle genes</p></td>
      </tr><tr><td>
          <p><code><a href="transcripts_to_genes.html">transcripts_to_genes()</a></code> </p>
        </td>
        <td><p>Ensembl Transcript Ids to Gene Symbols</p></td>
      </tr><tr><td>
          <p><code><a href="unite_metadata.html">unite_metadata()</a></code> </p>
        </td>
        <td><p>Unite metadata</p></td>
      </tr><tr><td>
          <p><code><a href="update_exp_meta.html">update_exp_meta()</a></code> </p>
        </td>
        <td><p>Update experiment Metadata</p></td>
      </tr><tr><td>
          <p><code><a href="update_human_gene_symbols.html">update_human_gene_symbols()</a></code> </p>
        </td>
        <td><p>Update human gene symbols in seurat object</p></td>
      </tr><tr><td>
          <p><code><a href="update_project_db.html">update_project_db()</a></code> </p>
        </td>
        <td><p>Update a database of seuratTools projects</p></td>
      </tr><tr><td>
          <p><code><a href="update_seu_meta.html">update_seu_meta()</a></code> </p>
        </td>
        <td><p>Update Seurat Metadata from Project</p></td>
      </tr><tr><td>
          <p><code><a href="update_seuratTools_object.html">update_seuratTools_object()</a></code> </p>
        </td>
        <td><p>Update a seuratTools Object</p></td>
      </tr><tr><td>
          <p><code><a href="velocyto_assay.html">velocyto_assay()</a></code> </p>
        </td>
        <td><p>run velocyto on a gene or transcript level seurat object</p></td>
      </tr><tr><td>
          <p><code><a href="velocyto_seu.html">velocyto_seu()</a></code> </p>
        </td>
        <td><p>run velocyto on a seurat object</p></td>
      </tr><tr><td>
          <p><code><a href="velocyto_seurat_from_loom.html">velocyto_seurat_from_loom()</a></code> </p>
        </td>
        <td><p>Run RNA Velocity starting with only a loom File</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Kevin Stachelek, Bhavana Bhat.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

